lib/inspec/plugin/v2/installer.rb

```diff
@@ -368,9 +368,19 @@ def install_gem_to_plugins_dir(new_plugin_dependency, # rubocop: disable Metrics
       # of the same gem on the stack or a malformed dependency.
       begin
         solution.each do |activation_request|
-          unless activation_request.full_spec.activated?
-            activation_request.full_spec.activate
+          requested_gemspec = activation_request.full_spec
+          next if requested_gemspec.activated?
+
+          gem_name = requested_gemspec.name
+          loaded_gem = Gem.loaded_specs[gem_name]
+          if loaded_gem
+            if requested_gemspec.version > loaded_gem.version
+              Gem.loaded_specs.delete(gem_name)
+            else
+              next # don't activate requested gemspec
+            end
```

Comment:
Not an important change, but to improve readability, we could consider refactoring something like:
```ruby
if loaded_gem && requested_gemspec.version <= loaded_gem.version
  next # Skip activation if the requested version is not newer
end

Gem.loaded_specs.delete(gem_name) if loaded_gem

```

Reply:
I guess you would like the refactoring done here https://github.com/inspec/inspec/pull/7259